File: Programming EF Code First/chapter6/Chapter6/Chapter 6 VB/DataAnnotations/Model/Trip.vb
Imports System.ComponentModel.DataAnnotations

Namespace Model
  Public Class Trip
	<Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)>
	Public Property Identifier() As Guid
	Public Property StartDate() As Date
	Public Property EndDate() As Date
	Public Property CostUSD() As Decimal
	<Timestamp>
	Public Property RowVersion() As Byte()

	Public Property Activities() As List(Of Activity)
  End Class
End Namespace

File: Programming EF Code First/chapter6/Chapter6/BAGA_DataAnnotations/Model/Trip.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;

namespace Model
{
  public class Trip
  {
    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Identifier { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public decimal CostUSD { get; set; }
    [Timestamp]
    public byte[] RowVersion { get; set; }

    public List<Activity> Activities { get; set; }
  }
}